tests/zfs-tests/tests/functional/fault/auto_spare_001_pos.ksh

```diff
@@ -0,0 +1,145 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+#
+# Copyright (c) 2017 by Intel Corporation. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/fault/fault.cfg
+
+#
+# DESCRIPTION:
+# Testing Fault Management Agent ZED Logic - Automated Auto-Spare Test.
+#
+# STRATEGY:
+# 1. Create a raidz pool with hot spares
+# 2. Write multiple files to the pool
+# 3. Export the pool to clear the cache
+# 4. Import the pool
+# 5. Configure zinject to cause read io errors on the first drive in the pool
```

Comment:
There are a couple slightly different cases and I think we want to test them all.

* `zpool offline -f <pool> <vdev>` is a great way to test basic sparing.
* `zinject -e io` tests that ZED retire agent proactively kicks in a spare after IO errors.
* `zinject -e checksum` tests that ZED retire agent proactively kicks in a spare after checksum errors.